contract ExceptionDefinition {
    exception X();
    exception Y(uint aug1, uint aug2);

    void foo() throws (X) {
        try {
            bar(1, 2);
        } catchall (Y expY) {
            throw X();
        }
    }

    void bar(uint a, uint b) throws (Y) {
        if (a != b) {
            throw Y(a, b);
        }
    }
}